accountability/heightlogs.go

```diff
@@ -0,0 +1,63 @@
+package accountability
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/mikanikos/Fork-Accountability/common"
+)
+
+// HeightLogs contains all messages for all the rounds from each process in a specific height
+type HeightLogs struct {
+	logs  map[string]*common.HeightVoteSet
+	mutex sync.RWMutex
+}
+
+// NewHeightLogs creates a new HeightLogs structure
+func NewHeightLogs() *HeightLogs {
+	return &HeightLogs{
+		logs: make(map[string]*common.HeightVoteSet),
+	}
+}
+
+// AddHvs adds a new hvs in the height HeightLogs
+func (hl *HeightLogs) AddHvs(processID string, hvs *common.HeightVoteSet) {
+	hl.mutex.Lock()
+	defer hl.mutex.Unlock()
+	hl.logs[processID] = hvs
```

Comment:
Yes, it is possible that the hvs already exists but this can only happen if the validators configs are initialized with non-unique ids. However, I was still handling this case by checking whether the map already contained the same value (there was a Contains method for that) and, if yes, I was not considering the hvs received (by the way, that's one of the many cases where a timeout for the monitor could be useful: if a wrong or non-valid hvs is sent, the algorithm might not be able to complete despite the theoretical correctness). Now, I merged the two methods for simplicity and use a LoadOrStore but the behavior is the same and should not give problems.